velox/exec/Window.cpp

```diff
@@ -374,19 +384,55 @@ void Window::callApplyForPartitionRows(
     // as WindowFunction only sees one partition at a time.
     rawPeerStarts[j] = peerStartRow_ - firstPartitionRow;
     rawPeerEnds[j] = peerEndRow_ - 1 - firstPartitionRow;
+  }
 
-    for (auto w = 0; w < numFuncs; w++) {
-      auto frameEndPoints = findFrameEndPoints(
-          w,
-          firstPartitionRow,
-          lastPartitionRow,
-          peerStartRow_,
-          peerEndRow_ - 1,
-          i);
-      rawFrameStartBuffers[w][j] = frameEndPoints.first - firstPartitionRow;
-      rawFrameEndBuffers[w][j] = frameEndPoints.second - firstPartitionRow;
+  auto updateFrameBounds = [&](vector_size_t* frameBound,
+                               core::WindowNode::BoundType boundType,
+                               core::WindowNode::WindowType type,
+                               bool isStartBound) -> void {
+    switch (boundType) {
+      case core::WindowNode::BoundType::kUnboundedPreceding:
+        findRowUnboundedPreceding(frameBound, numRows);
+        break;
+      case core::WindowNode::BoundType::kCurrentRow: {
+        if (type == core::WindowNode::WindowType::kRange) {
+          if (isStartBound) {
+            findCurrentRowFrameStart(
```

Comment:
The underlying logic of findCurrentRowFrameStart and findCurrentRowFrameEnds are very similar, except using peerStarts or peerEnds buffer. I think you can use a single function for both these and only pass the correct buffer based on isStartBound vs not.

Reply:
Thanks for the suggestion, I had retained the functions since the conditional check on isStartBound would still be needed, modified accordingly!